extras/kms/examples/config.go

```diff
@@ -0,0 +1,86 @@
+package examples
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/hashicorp/go-kms-wrapping/v2/aead"
+	"github.com/hashicorp/go-kms-wrapping/wrappers/transit/v2"
+	"github.com/hashicorp/go-secure-stdlib/configutil/v2"
+)
+
+// RootWrapperFromConfig returns the root wrapper from the provided kms hcl
+func RootWrapperFromConfig(ctx context.Context, kmsHcl string, useTransit bool) (wrapping.Wrapper, error) {
+	parsedKmsConfig, err := parseConfig(kmsHcl)
+	if err != nil {
+		return nil, err
+	}
+	var rootWrapper wrapping.Wrapper
+	switch {
+	case useTransit:
```

Comment:
This pattern is used twice in these examples, is it really preferable to an `if` `else`? Do we expect to add more cases?

Reply:
It might just be a preference thing, I know I find a switch like this to be more readable then an `if/else`.